src/main/java/com/aws/greengrass/certificatemanager/certificate/CISShadowMonitor.java

```diff
@@ -243,20 +247,37 @@ private synchronized void handleNewCloudVersion(int newVersion) {
                 for (CertificateGenerator cg : monitoredCertificateGenerators) {
                     cg.generateCertificate(connectivityInfoProvider::getCachedHostAddresses);
                 }
-                reportVersion(version);
-                lastVersion = version;
+
+                try {
+                    reportVersion(version);
+                } catch (ExecutionException e) {
+                    Throwable cause = e.getCause() == null ? e : e.getCause();
+                    LOGGER.atWarn().kv("version", version).cause(cause).log("Unable to report CIS shadow version");
+                } catch (TimeoutException e) {
+                    LOGGER.atWarn().kv("version", version).log("Timed out while reporting CIS shadow version");
+                } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)

Reply:
You haven't dealt with this exception correctly @jcosentino11, you must re-interrupt the thread using `Thread.currentThread().interrupt()`. Thread interruption means that someone wants your thread to shutdown, and catching the interrupted exception clears the thread interrupt flag. You must reset the flag.

Reply:
Will add---I think my confusion is that the interruption is happening within a completablefuture, which won't get interrupted on cancel like a regular future, so wasn't actually sure how that code path would happen.

Reply:
A Completable future is a future, and calling `.cancel(true)` will attempt to interrupt the thread running the task.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#cancel-boolean-
implementation is different, it'll complete the future with a cancellation exception

Reply:
The future completes with that exception, but the thread running the future is interrupted. This code here will be interrupted because it is waiting for a different future to complete.

Reply:
im probably overthinking this 😓 ,  regardless, this catch clause is now removed since the publish get call was removed 😄